lib/ansible/utils/doc_helpers.py

```diff
@@ -0,0 +1,23 @@
+# Copyright: (c) 2018, Dag Wieers <dag@wieers.com>
```

Comment:
Can we make this private (prepend the filename with a leading underscore)?  The reason is that it is only used at buildtime by our build scripts.  It isn't used at runtime at all.  Marking it private clues other people in to the fact that they shouldn't use it in runtime code.

Reply:
Well, I would prefer that if this functionality *is* useful, it does get used in runtime code, rather than being reinvented. That's why I would prefer it in a normal library, helper.py was a perfect fit IMO. But announcing it as "not to be used" seems to be the wrong reflex ?

Reply:
There's more code that is being used at build-time coming from the normal libraries.

Reply:
Do you have a use  case in the runtime code in mind that you can point me at?

Reply:
and yes, build time using run time code is fine.  What's misorganized is when build time uses runtime code which nothing in runtime uses.

Reply:
Private normally means 'not exposed for external use', because you want to avoid people to use it. I don't see why this function should not be used. This is another mental exercise and very ambiguous. Should we avoid people using *this* function, is there a greater harm that I am not seeing or should we encourage to reuse it if it is useful ? And why make an exceptional case for one simple function.

Currently we have utils/plugins_docs.py which is not private either, contains 2 functions that are only used by the same scripts. So I think you all have too much time on your hands to split hairs for things that don't matter as much as you think they do. Maybe, you can do those cleanups in a separate PR and merge this. Because I have had it with those small pet-peeves. I'd like to spend my time on more important things, if you don't mind :-)

Reply:
plugin_docs is an example of something that should not have been committed into utils without more thought.... however, if you ore closely, get_docstring() is in multiple files in runtime.  it uses add_fragments.  add_fragments use merge_fragment.  So everything in utils/plugin_docs is used by runtime.

If we have a runtime use case for this, then I see value in making sure it's a good api and looking at making it good for general usage.  If we don't have a runtime usage then we end up having to maintain an api which may not be well suited to the task people put it too.  that's why things not used at runtime should remain private until a use case emerges.  Then we can make sure the API is sufficient for the use cases and make it public.